daft/expressions/expressions.py

```diff
@@ -1098,3 +1103,33 @@ def iceberg_truncate(self, w: int) -> Expression:
             Expression: Expression of the Same Type of the input
         """
         return Expression._from_pyexpr(self._expr.partitioning_iceberg_truncate(w))
+
+
+class ExpressionJsonNamespace(ExpressionNamespace):
+    def query(self, jq_query: str) -> Expression:
+        """Query JSON data in a column using a JQ stylefilter https://jqlang.github.io/jq/manual/
+        This expression uses jaq as the underlying executor, see https://github.com/01mf02/jaq for the full list of supported filters.
+
+        Example:
+            >>> df = daft.from_pydict({"col": ['{"a": 1}', '{"a": 2}', '{"a": 3}']})
+            >>> df.with_column("res", df["col"].json.query(".a")).collect()
+            ╭──────────┬──────╮
+            │ col      ┆ res  │
+            │ ---      ┆ ---  │
+            │ Utf8     ┆ Utf8 │
+            ╞══════════╪══════╡
+            │ {"a": 1} ┆ 1    │
+            ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
+            │ {"a": 2} ┆ 2    │
+            ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
+            │ {"a": 3} ┆ 3    │
+            ╰──────────┴──────╯
+
+        Args:
+            jq_query (str): JQ query string
+
+        Returns:
+            Expression: Expression with the result of the JQ query
```

Comment:
Let's make it clear also that the results of the query is going to be a JSON-compatible string by default